### PR TITLE
DP-3718 : Added calculating for retrying time and added information abou...

### PR DESCRIPTION
**Pull request by endrius199202**
...t successful subscription or error

### CODE REVIEW

File: api/ws/connector.js [hunk 1]
```diff
@@ -41,6 +51,17 @@ function Websockets(conf, logger) {
     me.logger = logger;
     me.bindings = {};
 
+    me.calculateRetryTime = function(error) {
+        var time = utils.getMinutesFromMiliseconds(me.retryTime);
+        if(time.m >= 10) { // 10 minutes
+            me.retryTime = 1000 * 60 * 10;
```
Comment (adrianweber) [inline api/ws/connector.js]: retry time configuration should be define in config file (global.json)

File: api/ws/connector.js [hunk 2]
```diff
@@ -95,17 +115,30 @@ function Websockets(conf, logger) {
...
+                setTimeout(function() {
+                    me.reconnect();
+                }, parseInt(me.retryTime));
             });
             connection.on('message', function(message) {
-                me.logger.info('Fired STATUS: ', message.utf8Data);
-                me.onMessage(message.utf8Data);
+                parseMessage(message.utf8Data, function(err, messageObject) {
+                    if(!err) {
+                        if(messageObject.code === 200) {
```
Comment (adrianweber) [inline api/ws/connector.js]: define all error/success codes in separate file as constants. So you won't compare messageObject.code with magic numbers

File: api/ws/connector.js [hunk 3]
```diff
@@ -95,17 +115,30 @@ function Websockets(conf, logger) {
...
+                            me.retryTime = conf.connector.ws.retryTime;
+                            me.logger.info('WSConnector: Connection successful to: ' + conf.connector.ws.host + ':' + conf.connector.ws.port);
+                        } else if(messageObject.code === 1024) {
+                            me.logger.info('Fired STATUS: ', messageObject.content);
+                            me.onMessage(messageObject.content);
+                        } else if(messageObject.code === 500 || messageObject.code === 401) {
+                            me.calculateRetryTime(messageObject.content);
+                        }
+                    }
```
Comment (adrianweber) [inline api/ws/connector.js]: there is missing else clause(for case when err !== null). Add it with logger.error message

File: api/ws/connector.js [hunk 4]
```diff
@@ -78,15 +100,14 @@ function Websockets(conf, logger) {
 
     me.onMessage = function(message) {
         try {
-            me.bindings.handler(JSON.parse(message));
+            me.bindings.handler(message);
```
Comment (mtomczew) [inline api/ws/connector.js]: This won't throw any exception. I'll remove try {} catch () {}

Comment (mtomczew) [inline api/ws/connector.js]: Done.

File: lib/utils.js [hunk 5]
```diff
@@ -156,6 +156,12 @@ IoTKitUtils.prototype.getValueFromDeviceConfig = function(key) {
     return me.deviceConf[key];
 };
 
+IoTKitUtils.prototype.getMinutesFromMiliseconds = function(miliseconds) {
```
Comment (mtomczew) [inline lib/utils.js]: I'll change function name.

**Comment by mtomczew** [comment]
Merged

**Pull request closed without merging**